servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RepeatStrategies.java

```diff
@@ -223,7 +223,7 @@ private RepeatStrategies() {
         final long maxDelayNanos = maxDelay.toNanos();
         final long maxInitialShift = maxShift(initialDelayNanos);
         return repeatCount -> {
-            final long baseDelayNanos = initialDelayNanos << min(maxInitialShift, repeatCount - 1);
+            final long baseDelayNanos = min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, repeatCount - 1));
```

Comment:
can you extract this out into a shared method? looks like we had this in other methods but it was missed in a few spots.

```diff
diff --git a/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RepeatStrategies.java b/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RepeatStrategies.java
index 99705c6df..6124e6a50 100644
--- a/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RepeatStrategies.java
+++ b/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RepeatStrategies.java
@@ -19,6 +19,7 @@ import java.time.Duration;
 import java.util.function.IntFunction;
 
 import static io.servicetalk.concurrent.api.Completable.failed;
+import static io.servicetalk.concurrent.api.RetryStrategies.baseDelayNanos;
 import static io.servicetalk.concurrent.api.RetryStrategies.checkJitterDelta;
 import static io.servicetalk.concurrent.api.RetryStrategies.checkMaxRetries;
 import static io.servicetalk.concurrent.api.RetryStrategies.maxShift;
@@ -166,7 +167,7 @@ public final class RepeatStrategies {
         final long maxDelayNanos = maxDelay.toNanos();
         final long maxInitialShift = maxShift(initialDelayNanos);
         return repeatCount -> timerExecutor.timer(current().nextLong(0,
-                        min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, repeatCount - 1))), NANOSECONDS);
+                        baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, repeatCount)), NANOSECONDS);
     }
 
     /**
@@ -196,7 +197,7 @@ public final class RepeatStrategies {
         final long maxInitialShift = maxShift(initialDelayNanos);
         return repeatCount -> repeatCount <= maxRepeats ?
                 timerExecutor.timer(current().nextLong(0,
-                        min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, repeatCount - 1))), NANOSECONDS) :
+                        baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, repeatCount)), NANOSECONDS) :
                 terminateRepeat();
     }
 
@@ -223,7 +224,7 @@ public final class RepeatStrategies {
         final long maxDelayNanos = maxDelay.toNanos();
         final long maxInitialShift = maxShift(initialDelayNanos);
         return repeatCount -> {
-            final long baseDelayNanos = min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, repeatCount - 1));
+            final long baseDelayNanos = baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, repeatCount);
             return timerExecutor.timer(
                     current().nextLong(max(0, baseDelayNanos - jitterNanos),
                             min(maxDelayNanos, addWithOverflowProtection(baseDelayNanos, jitterNanos))),
@@ -261,7 +262,7 @@ public final class RepeatStrategies {
             if (repeatCount > maxRepeats) {
                 return terminateRepeat();
             }
-            final long baseDelayNanos = min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, repeatCount - 1));
+            final long baseDelayNanos = baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, repeatCount);
             return timerExecutor.timer(
                     current().nextLong(max(0, baseDelayNanos - jitterNanos),
                             min(maxDelayNanos, addWithOverflowProtection(baseDelayNanos, jitterNanos))),
diff --git a/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RetryStrategies.java b/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RetryStrategies.java
index ff089ca75..16f878a6d 100644
--- a/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RetryStrategies.java
+++ b/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RetryStrategies.java
@@ -170,7 +170,7 @@ public final class RetryStrategies {
         final long maxInitialShift = maxShift(initialDelayNanos);
         return (retryCount, cause) -> causeFilter.test(cause) ?
                 timerExecutor.timer(current().nextLong(0,
-                        min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, retryCount - 1))), NANOSECONDS) :
+                        baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, retryCount)), NANOSECONDS) :
                 failed(cause);
     }
 
@@ -204,7 +204,7 @@ public final class RetryStrategies {
         final long maxInitialShift = maxShift(initialDelayNanos);
         return (retryCount, cause) -> retryCount <= maxRetries && causeFilter.test(cause) ?
                 timerExecutor.timer(current().nextLong(0,
-                        min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, retryCount - 1))), NANOSECONDS) :
+                        baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, retryCount)), NANOSECONDS) :
                 failed(cause);
     }
 
@@ -237,7 +237,7 @@ public final class RetryStrategies {
             if (!causeFilter.test(cause)) {
                 return failed(cause);
             }
-            final long baseDelayNanos = min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, retryCount - 1));
+            final long baseDelayNanos = baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, retryCount);
             return timerExecutor.timer(
                     current().nextLong(max(0, baseDelayNanos - jitterNanos),
                             min(maxDelayNanos, addWithOverflowProtection(baseDelayNanos, jitterNanos))),
@@ -278,7 +278,7 @@ public final class RetryStrategies {
             if (retryCount > maxRetries || !causeFilter.test(cause)) {
                 return failed(cause);
             }
-            final long baseDelayNanos = min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, retryCount - 1));
+            final long baseDelayNanos = baseDelayNanos(maxDelayNanos, initialDelayNanos, maxInitialShift, retryCount);
             return timerExecutor.timer(
                     current().nextLong(max(0, baseDelayNanos - jitterNanos),
                             min(maxDelayNanos, addWithOverflowProtection(baseDelayNanos, jitterNanos))),
@@ -302,4 +302,8 @@ public final class RetryStrategies {
     static long maxShift(long v) {
         return numberOfLeadingZeros(v) - 1;
     }
+
+    static long baseDelayNanos(long maxDelayNanos, long initialDelayNanos, long maxInitialShift, int count) {
+        return min(maxDelayNanos, initialDelayNanos << min(maxInitialShift, count - 1));
+    }
 }

```